src/config/models.rs

```diff
@@ -292,6 +305,7 @@ impl ConfigOpts {
             (None, None) => None,
             (Some(val), None) | (None, Some(val)) => Some(val),
             (Some(l), Some(mut g)) => {
+                g.watch = g.watch.or(l.watch);
                 g.ignore = g.ignore.or(l.ignore);
                 Some(g)
             }
```

Comment:
Very nice. Hopefully this layered config system wasn't too complex to grok at first. All of your updates here look correct!